pandas/core/indexes/multi.py

```diff
@@ -235,16 +235,28 @@ def set_levels(self, levels, level=None, inplace=False,
                    labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
                    names=[u'foo', u'bar'])
         """
-        if level is not None and not is_list_like(level):
-            if not is_list_like(levels):
-                raise TypeError("Levels must be list-like")
-            if is_list_like(levels[0]):
-                raise TypeError("Levels must be list-like")
+
+        level_list_like = level is None or is_list_like(level)
+        levels_list_like = (is_list_like(levels) and
```

Comment:
further some of this validation is duplicative of ``_set_levels``. So rather add any *additional* validation there directly.